### PR TITLE
refactor: DiagnosticUsage uses the built-in scope

### PR DESCRIPTION
**Pull request by thermokarst**

### CODE REVIEW

File: qiime2/sdk/tests/test_usage.py [hunk 1]
```diff
@@ -31,67 +31,83 @@ def test_basic(self):
...
+        self.assertEqual('init_data', obs2.source)
+        self.assertEqual('init_data', obs3.source)
+        self.assertEqual('comment', obs4.source)
+        self.assertEqual('action', obs5.source)
+        self.assertEqual('comment', obs6.source)
+        self.assertEqual('action', obs7.source)
+
+        self.assertTrue('chained usage (pt 1)' in obs4.result['text'])
+
+        self.maxDiff = None
```
Comment (thermokarst) [inline qiime2/sdk/tests/test_usage.py:78]: TODO: remove this

File: qiime2/sdk/usage.py [hunk 2]
```diff
@@ -637,7 +637,8 @@ def _get_metadata_column_(self, column_name, record):
         raise NotImplementedError
 
     def comment(self, text: str):
-        return self._comment_(text)
+        comment = self._comment_(text)
+        return self._push_record(str(comment), comment, 'comment')
```
Comment (thermokarst) [inline qiime2/sdk/usage.py:641]: This was an oversight - comments should generate scope records.

File: qiime2/sdk/usage.py [hunk 3]
```diff
@@ -738,69 +739,67 @@ class DiagnosticUsage(Usage):
     """
     def __init__(self):
         super().__init__()
-        self.recorder = []
```
Comment (thermokarst) [inline qiime2/sdk/usage.py:741]: This was leftover from the design process - the out-of-band recorder shouldn't be necessary - Scope should handle everything.

**Review by andrewsanchez — APPROVED** [review]
Looks good to me!  🔍

**Pull request merged**